internal/metrics/dashboard/dashboards.go

```diff
@@ -83,14 +87,25 @@ func ReconcileForDashboards(k8sClient client.Client, reader client.Reader) error
 }
 
 // RemoveDashboardConfigMap removes the config map in the grafana dashboard
-func RemoveDashboardConfigMap(c client.Client) (err error) {
-	cm := &corev1.ConfigMap{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      DashboardName,
-			Namespace: DashboardNS,
-		},
+func RemoveDashboardConfigMap(c client.Client, r client.Reader) (err error) {
+	cm := newDashboardConfigMap()
+	s := labels.NewSelector()
+	for k, v := range cm.Labels {
+		r, _ := labels.NewRequirement(k, selection.Equals, []string{v})
+		s = s.Add(*r)
+	}
+	options := &client.ListOptions{
+		LabelSelector: s,
+		Namespace:     DashboardNS,
 	}
-	return c.Delete(context.TODO(), cm)
+	configMaps := &corev1.ConfigMapList{}
+	if err := r.List(context.TODO(), configMaps, options); err != nil {
```

Comment:
How about Get(cm) and then compare the actual labels with the expected ones? 
The List() could find configmaps that have identical labels but a different name - its unlikely, but a Get is more efficient and direct.

Reply:
@alanconway updated to check labels